### ISSUE TITLE
no longer showing enemybuffs

### ISSUE THREAD

**Issue by daswb** [rgd87/ClassicAuraDurations#21]
CAD enemybuffs doesn't seem to be working at the moment. Getting reports from all my friends/discords of the same issue.

I'm on SOD and have tried the following:

-Full aura replacement is checked
-typed /cad enemybuffs
-reinstalled both cad and libclassicdurations
-trying old versions
-trying wotlk versions
-trying old wotlk versions

This addon has been amazing for years. Thanks for looking into it!

**Comment by d87** [comment]
blizzard fixed the loophole that allowed it to work so well

**Comment by Aleksxz** [comment]
It did kinda work just before the new update this week, you could see some vanilla buff on target, but now there is nothing working anymore. I'm sure there is something to do.

**Comment by d87** [comment]
I did, in the latest CAD version, enemy buffs applied in your combat log range should be displayed, but stuff like world buffs is no longer possible

**Comment by Aleksxz** [comment]
Thank you for the work, but for me and everyone else I know, we still can't see enemybuffs, even after reinstalling and removing savedvariables, and typing /Cad enemybuffs ofc.

**Comment by Andrew-Cha** [comment]
Hi!

The `Libs` folder is out of date. [This commit you've added](https://github.com/rgd87/LibClassicDurations/commit/4775838d442532e866c8ef045eea0f1b0cf879df) seems to be a partial workaround in my personal testing.

How are `Libs` folders updated in WoW addon development?

Could you clarify what exactly Blizzard did to break previous functionality? I could push out a PR. 

Thanks.

**Comment by d87** [comment]
Ah crap, it's pulling from curseforge, not github... I forgot all about it. 
There was a bug in tooltip functions that allowed to list names of enemy buffs without restriction. Now it's fixed as I'm told
CAD 3.4.4 should have the updated lib version

**Comment by Hmmintriguing** [comment]
Would it be possible to collaborate with the creator of Rogue UI (spelled RougeUI) in regards to getting enemy buffs to work properly for Classic Aura Duration?

They have enemy buffs working terrifically and in their latest update, which was yesterday, they added the tooltips to enemy buffs as well.

The only drawback is in order for enemy buffs to be seen, you MUST use their unitframes as well. I tried testing enemy buffs by turning off their unitframes and using Luna Unit Frames instead, but it was a no-go unfortunately.

Not sure how willing they would be willing to work with you or anyone for that matter to get enemy buffs working again for CAD or any other addon, but I suppose it would be worth a shot.

Here is a link to their addon: https://legacy.curseforge.com/wow/addons/rougeui/files/5193043

**Comment by d87** [comment]
At a glance he didn't invent any hidden black magic, and is just using LCD

**Comment by Hmmintriguing** [comment]
To be frank, I would much rather use CAD and have my own unitframe setup, rather than be forced to use a specific unitframe that I do not like in order to see enemy buffs.

Would you be able to mimic their fixes from their addon and consider applying them to CAD?

**Comment by naroca050** [comment]
In one of his [latest videos](https://www.youtube.com/watch?v=jGKlpePNmqU), Pshero is shown to have an addon like CAD that seems to work relatively fine. Is it CAD? Because to me buffs don't show up like this when others activate buffs close to me.

**Comment by Hmmintriguing** [comment]
> In one of his [latest videos](https://www.youtube.com/watch?v=jGKlpePNmqU), Pshero is shown to have an addon like CAD that seems to work relatively fine. Is it CAD? Because to me buffs don't show up like this when others activate buffs close to me.

Yes, he has CAD. In his video, you can tell because, although it is working to an extent, it is not tracking the majority of buffs. The paladin, druid, and warrior all should have had paladin buffs & druid buffs showing, but as you can see, they weren't shown on his default Blizzard unit frames.

The only working CAD-like addon there is that is working currently is the one I mentioned here, which is [RougeUI](https://www.curseforge.com/wow/addons/rougeui). Unfortunately, you are forced to use their unit frames in order for the majority of enemy buffs to be shown (from my limited testing, it does not track world buffs).

I was hoping that @d87 would be able to contact & collaborate with the creator of RougeUI to get CAD functioning at it's full potential again, but it doesn't look like that will be the case anytime soon or at all unfortunately.

**Comment by naroca050** [comment]
Perhaps contacting the creator of RougeUI to ask him to try and implement it using the default frames could be fruitful? I hope that CAD is not abandoned. It's very needed for PvP in particular

**Comment by XyzKangUI** [comment]
> At a glance he didn't invent any hidden black magic, and is just using LCD

It actually does have hidden black magic which got patched in 1.15.2. Going forward I am going to use LCD 😄

**Comment by Hmmintriguing** [comment]
> Going forward I am going to use LCD 😄

Are your referring to [LibClassicDurations?](https://www.curseforge.com/wow/addons/libclassicdurations) If so, this hasn't been updated since April of 2021.

**Comment by XyzKangUI** [comment]
> > Going forward I am going to use LCD 😄
> 
> Are your referring to [LibClassicDurations?](https://www.curseforge.com/wow/addons/libclassicdurations) If so, this hasn't been updated since April of 2021.

Yes, once patch 1.15.2 releases RougeUI will be using [LCD ](https://github.com/rgd87/LibClassicDurations) which was updated one month ago.

At the moment RougeUI is capable of seeing all the enemy buffs by using Blizz's API, which has been permanently fixed in 1.15.2. 

After five years, the Blizzard Classic team suddenly decided to patch the ability to view enemy buffs in Classic. The Classic userbase, accustomed to seeing enemy buffs, now considers CAD (ClassicAuraDurations) broken for not displaying buffs applied outside the combat log range. Furthermore, the fact that CAD only shows auras existing in its spell table contributes to the idea of it being broken in 1.15.x, because the spell table doesn't contain newer (rune) spells.

That said, I'm puzzled by Blizzard's decision to go against the community, particularly during Season of Discovery, which significantly deviates from Vanilla. It's a modern take on Vanilla that should still provide the ability to view enemy buffs, as users have become accustomed to in every version since Vanilla, including every Classic version for the past five years.

**Comment by Hmmintriguing** [comment]
Ah, okay. You're the creator of RougeUI.

Apologies, but where are you seeing LCD being updated one month ago? I checked Curseforge & the GitHub, but like I mentioned, there have been no new releases since 2021. Is there something I am missing, because if LCD is still able to track enemy buffs as of right now, I will use that instead of CAD.

Also, is there any way to use a different unit frame addon in conjunction with RogueUI's ability to see enemy buffs, as I am not fond of using Blizzard's default unit frames? If not, regardless, I'm just glad there is still an addon that has the functionality to track enemy buffs on servers that run Classic Vanilla (even though SOD is nothing like Vanilla).

**Comment by XyzKangUI** [comment]
> Is there something I am missing, because if LCD is still able to track enemy buffs as of right now, I will use that instead of CAD.

The latest commit on LCD's GitHub is from one month ago. It's important to note that you won't see all enemy buffs, unlike before the tooltip scan exploit was patched or currently with RougeUI. You'll only see buffs that have been applied within the combat log range **if** they are listed in LCD's classAbilities.lua.

> Also, is there any way to use a different unit frame addon in conjunction with RogueUI's ability to see enemy buffs, as I am not fond of using Blizzard's default unit frames?

The short answer is no, but it won't matter soon because RougeUI won't be able to see enemy buffs in a few days.

From what I've read in your earlier messages, there appears to be some confusion. CAD exclusively functions with the default Blizzard frames. The compatibility of CAD with Luna frames is due to Luna's utilization of LibClassicDurations, which CAD adds by including a newer version of the libraries.
